### PR TITLE
Project selection view

### PR DESCRIPTION
**Pull request by taavirehemagi**
Closes https://github.com/toggl/mobile/issues/864

### CODE REVIEW

File: Joey/UI/Activities/ProjectListActivity.cs [hunk 1]
```diff
@@ -1,36 +1,175 @@
...
+        private class ProjectFragmentAdapter : FragmentPagerAdapter
+        {
+            private List<Fragment> fragments = new List<Fragment> ();
+            private List<String> fragmentTitles = new List<String> ();
+            private List<WorkspaceData> workspaces;
+            private int firstTabPos = 0;
+
+            public ProjectFragmentAdapter (FragmentManager fm, IList<TimeEntryData> timeEntryList) : base (fm)
+            {
+                workspaces = WorkspaceProjectsView.GetWorkspaces().Result;
```
Comment (monday8am) [inline Joey/UI/Activities/ProjectListActivity.cs]: This code "blocks" an async call. The order of how to create views and load data should be changed. The starting point is OnCreateActivity, then we should:

- Init native components.
- Parse data passed to activities.
- Load data from db using correct async calls.
- Feed native components with the data loaded.

File: Phoebe/Data/Views/WorkspaceProjectsView.cs [hunk 2]
```diff
@@ -521,19 +512,45 @@ public bool IsLoading
             }
         }
 
+        public static async Task<List<WorkspaceData>> GetWorkspaces()
```
Comment (monday8am) [inline Phoebe/Data/Views/WorkspaceProjectsView.cs]: GetWorkspaces should be GetWorkspacesAsync. 

This method is not used inside the class. Together with SortWorkspaces, the only purpose together with SortWorkspaces is to returns sorted workspaces to an external object. And it doesn't use any internal resource. 

We should move it to another place.

**Comment by monday8am** [comment]
Considering what we talked before, there is a change of approach. My suggestion is:
- Keep only one ProjectListViewModel like before.
- WorkspaceProjectsView is inside the view model and should have a workspace modifier. If you pass a workspace id the content should be change.
- The workspace change should be triggered by **TabLayout.OnTabSelectedListener**

**Comment by taavirehemagi** [comment]
For now, my solution would be do keep the approach and pager the same, just disable the swiping. In my opinion, this approach keeps the animation of views sliding and the models are more easy and smaller. You feel strongly about having one ProjectListModel? Workspace-based is good for now in my opinion, since we're not sure how it will look/be used in iOS. 

Also, can you check the state now? I think I fixed what you told me.

**Comment by monday8am** [comment]
@taavirehemagi 
> You feel strongly about having one ProjectListViewModel
We must have a ProjectListViewModel because is the only way it makes sense. In the ViewModel we should have represented everything that appears in the view. 

- If you keep the viewPages, then you have to keep different lists. This is more complex than one list.
- And you show to the users that the lists can be moved (right - left). But they can't moved them. Is confused. Is the problem of use half functionality in a component.

**Comment by taavirehemagi** [comment]
aalright :+1:

**Comment by taavirehemagi** [comment]
I'm running out of clever things to do here... :) Want to check? @monday8am

**Pull request merged**